### PR TITLE
Adapt alias to handle de coc

### PR DESCRIPTION
**Pull request by mestachs**
## Self proof reading checklist

- [ ] Did I run Pronto and fixed all warnings
- [ ] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [ ] Did I test the right thing?
- [ ] Did I test corner cases, non happy path (defensive testing)?
- [ ] Check that I used the ad-hoc patterns and created files accordingly
- [ ] Check code efficiency with record intensive project
- [ ] Update documentation,readme accordingly

Thanks!

### CODE REVIEW

File: spec/coc_spec.rb [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            "period"              => "2018Q1",
+            "dataElement"         => "dhis2acta",
+            "categoryOptionCombo" => "stockendcoc",
+            "value"               => "6"
+          }
+        ]
+      )
+    end
+
+    it "exports nil values to dhis2 when no dhis2 values" do
```
Comment (pjaspers) [inline spec/coc_spec.rb:206]: (I think that lines needs better words)

But this might also need a comment for future me, something like:

``` ruby
# Package A will read from `dhis2acta` got get the `stock_start`
# (`stockstartcoc`) and `stock_end` (`stockendcoc`), it will then
# output a `comsumption` to `"dhis2acta.consumptioncoc"`.
#
# Package B will read from `dhis2acta` to get the `stock_end`
# (`stockendcoc`) and from the newly created
# `dhis2acta.consumptioncoc`
#
# So putting this all together, if `start_stock` is 8 and `stock_end`
# is 6, the `consumption` will be 2 and that will be stored in
# `dhis2acta.consumptioncoc`
```

File: lib/orbf/rules_engine/builders/alias_post_processor.rb [hunk 2]
```diff
@@ -38,13 +38,17 @@ def call
...
-          data_element = variable.public_send dhis2_element_meth
+          data_element = variable.public_send(dhis2_element_meth)
+          coc = coc_meth ? variable.public_send(coc_meth) : "default"
           next unless data_element
+          components = data_element.split('.')
           key = { period:       variable.dhis2_period,
                   orgunit:      variable.orgunit_ext_id,
-                  data_element: data_element }
+                  data_element: components[0],
+                  coc: components[1] || coc || "default"
```
[the remark this answers is not in the document]
Comment (mestachs) [reply lib/orbf/rules_engine/builders/alias_post_processor.rb:50]: there's an asymetry between "inputs" and "outputs"

the first call : `hash_in = index_variables(:dhis2_in_data_element)`

  - dhis2_in_data_element return 'deid.cocid' or a 'deid'
  - components[1] might be filled, 
  - coc is nil

the second call ` hash_out = index_variables(:dhis2_data_element, :dhis2_coc)`

 - dhis2_data_element = 'deid'
 - coc = nil or 'cocid'

[the remark this answers is not in the document]
Comment (pjaspers) [reply lib/orbf/rules_engine/builders/alias_post_processor.rb:50]: Ah yes, the `coc` can always be `nil` so we still need to fallback to `default`. Got it.

**Comment by pjaspers** [comment]
Thanks @mestachs!
